### ISSUE TITLE
SIGSEGV due to static[T] overloading

### ISSUE THREAD

**Issue by GULPF** [nim-lang/Nim#6899]
```nim
proc foo*(x: string, y: int) =
    discard x & ";" # SIGSEGV: Illegal storage access.

proc foo*(x: static[string], y: static[int]) =
    discard

let nonstatic = 0
foo("abc", nonstatic)
```

If `foo` is invoked with both parameters as either static or non-static, it works. But if the parameters are mixed (one static, one non-static), bad things happen.

**Comment by krux02** [comment]
well I could create a new bug, but this is too much related:


```Nim

macro foobar(note: static[string] = ""): untyped =
  discard

foobar()

```

compilation output:

```
-*- mode: nim-compile; default-directory: "/tmp/" -*-
nim-compile started at Mon Dec 11 17:08:23

nim c -r /tmp/scratch2.nim
Hint: used config file '/local/doering/Nim/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: scratch2 [Processing]
SIGSEGV: Illegal storage access. (Attempt to read from nil?)

nim-compile exited abnormally with code 1 at Mon Dec 11 17:08:23

```

**Comment by GULPF** [comment]
Fixed by #8072 (both examples work)
